controllers/productController.js

```diff
@@ -1,13 +1,29 @@
 const productService = require('../services/productService');
-const { catchError } = require('../middlewares/error');
+const { catchError, baseError } = require('../middlewares/error');
 
 const getProductById = catchError(async (req, res) => {
   const { productId } = req.params;
-  const { mainCategory } = req.body;
-  const product = await productService.getProductById(productId, mainCategory);
+  if (!productId) {
+    return new baseError('NOT_PRODUCTID', 400);
+  }
+  const [product] = await productService.getProductById(productId);
   return res.status(200).json(product);
 });
 
+const categoryPage = catchError(async (req, res) => {
```

Comment:
해당 API는 category page에서만 쓰이는 것이 아닐 수 있습니다. 어디서든 가져다가 쓸수 있기 때문에 API의 함수명을 지을 때는 page가 아닌 데이터 중심으로 보셔야 합니다.
해당 API는 제품 목록을 가져오는 API 이므로 다음과 같이 함수 명을 수정해 주세요.
```suggestion
const getProductList = catchError(async (req, res) => {
```